### ISSUE TITLE
Panther doesn't start when launching PhpUnit Testsuite (but not when launching without testsuite)

### ISSUE THREAD

**Issue by bastien70** [symfony/panther#530]
## Versions
Symfony 6.0
PHP 8.1
PhpUnit 9.5.11
symfony/phpunit-bridge 6.0

## Description
Panther starts fine when running tests without a testsuite option. On the other hand, if we add a --testsuite option to our tests, then the Panther tests end in error

## Example

1. I run `symfony php vendor/bin/phpunit` which will run all tests (some unit tests, integration tests, EndToEnd tests with Panther) => No problem
2. I run a specific test or a class test or a directory test (with PhpStorm) with some differents tests and Panther tests => No error
3. I run the same directory tests but with the `--testsuite` option, like this : `symfony php vendor/bin/phpunit --testsuite application`, and the Panther tests result by an error.

## Error
When launching tests with `--testsuite` option, the first panther test error will be :

```
RuntimeException : Could not connect to web server after 30 seconds (Idle timeout reached for "http://127.0.0.1:9080/".). /home/bastien/Bureau/Symfony/wegical/vendor/symfony/panther/src/ProcessManager/WebServerReadinessProbeTrait.php:82
 /home/bastien/Bureau/Symfony/wegical/vendor/symfony/panther/src/ProcessManager/WebServerManager.php:84
 /home/bastien/Bureau/Symfony/wegical/vendor/symfony/panther/src/PantherTestCaseTrait.php:120
 /home/bastien/Bureau/Symfony/wegical/vendor/symfony/panther/src/PantherTestCaseTrait.php:163
 /home/bastien/Bureau/Symfony/wegical/tests/General/Controller/AbstractControllerTest.php:185
 /home/bastien/Bureau/Symfony/wegical/tests/Application/Controller/AbsenceControllerTest.php:322
```

And this error for next Panther's tests : 

```
Facebook\WebDriver\Exception\InvalidArgumentException : invalid argument
  (Session info: headless chrome=97.0.4692.71) /home/bastien/Bureau/Symfony/wegical/vendor/php-webdriver/webdriver/lib/Exception/WebDriverException.php:99
 /home/bastien/Bureau/Symfony/wegical/vendor/php-webdriver/webdriver/lib/Remote/HttpCommandExecutor.php:372
 /home/bastien/Bureau/Symfony/wegical/vendor/php-webdriver/webdriver/lib/Remote/RemoteWebDriver.php:585
 /home/bastien/Bureau/Symfony/wegical/vendor/php-webdriver/webdriver/lib/Remote/RemoteWebDriver.php:246
 /home/bastien/Bureau/Symfony/wegical/vendor/symfony/panther/src/Client.php:529
 /home/bastien/Bureau/Symfony/wegical/vendor/symfony/panther/src/Client.php:273
 /home/bastien/Bureau/Symfony/wegical/tests/General/Controller/AbstractControllerTest.php:100
 /home/bastien/Bureau/Symfony/wegical/tests/Application/Controller/AccountControllerTest.php:177
```

I don't understand at all how this is possible.

My panther tests will still be initialized the same way though.

```php
$client = static::createPantherClient(
            [
                '--window-size=1200,1100',
                '--no-sandbox',
                '--disable-gpu',
            ],
            [],
            [],
        );
        $this->client = $client;
```

And my **phpunit.xml.dist** file : 

```xml
<?xml version="1.0" encoding="UTF-8"?>

<!-- https://phpunit.readthedocs.io/en/latest/configuration.html -->
<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd"
         backupGlobals="false"
         colors="true"
         bootstrap="tests/bootstrap.php"
         convertDeprecationsToExceptions="false"
>
    <php>
        <ini name="display_errors" value="1" />
        <ini name="error_reporting" value="-1" />
        <server name="APP_ENV" value="test" force="true" />
        <server name="SHELL_VERBOSITY" value="-1" />
        <server name="SYMFONY_PHPUNIT_REMOVE" value="" />
        <server name="SYMFONY_PHPUNIT_VERSION" value="9.5" />
    </php>

    <testsuites>
        <testsuite name="Project Test Suite">
            <directory>tests</directory>
        </testsuite>
        <testsuite name="application">
            <directory>tests/Application/Command</directory>
            <directory>tests/Application/Entity</directory>
            <directory>tests/Application/Form</directory>
            <directory>tests/Application/Controller</directory>
        </testsuite>
    </testsuites>

    <coverage processUncoveredFiles="true">
        <include>
            <directory suffix=".php">src</directory>
        </include>
    </coverage>

    <listeners>
        <listener class="Symfony\Bridge\PhpUnit\SymfonyTestsListener" />
    </listeners>

    <extensions>
        <extension class="Symfony\Component\Panther\ServerExtension" />
        <extension class="DAMA\DoctrineTestBundle\PHPUnit\PHPUnitExtension" />
    </extensions>

</phpunit>

```

**Comment by domis86** [comment]
>    1. I run symfony php vendor/bin/phpunit which will run all tests (some unit tests, integration tests, EndToEnd tests with Panther) => No problem
>    2. I run a specific test or a class test or a directory test (with PhpStorm) with some differents tests and Panther tests => No error
>    3. I run the same directory tests but with the --testsuite option, like this : symfony php vendor/bin/phpunit --testsuite application, and the Panther tests result by an error.


Can you also try to run phpunit with "group" option? ( https://phpunit.readthedocs.io/en/9.5/annotations.html#group )
So assign some group to some test(s) and then run: `vendor/bin/phpunit --group=that_assigned_group_name`

**Comment by bastien70** [comment]
> Can you also try to run phpunit with "group" option? ( https://phpunit.readthedocs.io/en/9.5/annotations.html#group ) So assign some group to some test(s) and then run: `vendor/bin/phpunit --group=that_assigned_group_name`

Yes it's working.
I added the group option to several test classes (unit, integration, and panther).

If I run the `vendor/bin/phpunit --group=app` command

The tests are all carried out without error (but suddenly the order of passage is not respected, hence the idea of ​​--testsuite, but I cannot understand why this option is the problem )

**Comment by domis86** [comment]
> hence the idea of ​​--testsuite, but I cannot understand why this option is the problem

unfortunately me too :/

> but suddenly the order of passage is not respected

I know that sometimes order of execution of tests is annoying when is not an order which you want. Is possible to force your custom order of execution but it requires changing/hacking some phpunit classes - which probably you dont want to do. 

On the other hand: in theory tests should be isolated and work in any order (in other words: order of execution shouldnt matter and tests should be always green).

**Comment by bastien70** [comment]
Yes I know the tests should be able to run in any order. Except that I use DamaDoctrineTestBundle which allows me to rollback database transactions after each test, and it won't work for EndToEnd tests.

Which forces me to place all my unit and integration tests before EndToEnd tests.

And I have several folders each containing unit, integration, and EndToEnd tests, so I can't cheat by intentionally placing some folders before others in my project tree.

I have to use the testsuite.

Now, surely there must be an explanation why this option is problematic.

**Comment by domis86** [comment]
Not related to panther, but in such situation i used following solution:
1. assign one of the group for each test class: "unit", "integration", "end2end"
2. for each group run phpunit CLi command:

```
vendor/bin/phpunit --group=unit
vendor/bin/phpunit --group=integration
vendor/bin/phpunit --group=end2end
```

Then you will force at least that tests for specific groups are run before/after tests with other groups.

**Comment by bastien70** [comment]
Yes it's a good idea. Afterwards, it's a shame because it won't be done in a single order.

And above all, it will no longer be possible to launch them with PhpStorm helper, something that I found useful.

I imagine that this is the only solution to adopt for the moment, while waiting to understand why the --testsuite option gives us this kind of error

**Comment by bastien70** [comment]
Seems that this is not a problem relative to the `--testsuite` option.
In my phpunit.xml.dist, when configuring my testsuite, I have way more directories than what I showed above.

When I remove a directory that will take some time to test, then the Panther tests run.

In case Panther tests are run after some long time (after all previous tests) then they don't seem to run. This will loop 30 seconds on the 1st one it encounters, flag the error I quoted above, then open a completely blank browser for others, and error to console 'invalid argument'
